src/plugins/data/public/search/aggs/buckets/terms_fn.ts

```diff
@@ -0,0 +1,181 @@
+      }),
+    },
+    order: {
+      types: ['string'],
+      required: true,
+      help: i18n.translate('data.search.aggs.buckets.terms.order.help', {
+        defaultMessage: 'Order in which to return the results: asc or desc',
+      }),
+    },
+    orderBy: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.buckets.terms.orderBy.help', {
+        defaultMessage: 'Field to order results by',
+      }),
+    },
+    orderAgg: {
+      types: ['agg_type'],
+      help: i18n.translate('data.search.aggs.buckets.terms.orderAgg.help', {
+        defaultMessage: 'Agg config to use for ordering results',
+      }),
+    },
+    size: {
+      types: ['number'],
+      default: 5,
+      help: i18n.translate('data.search.aggs.buckets.terms.size.help', {
+        defaultMessage: 'Max number of buckets to retrieve',
+      }),
+    },
+    missingBucket: {
+      types: ['boolean'],
+      default: false,
+      help: i18n.translate('data.search.aggs.buckets.terms.missingBucket.help', {
+        defaultMessage: 'When set to true, groups together any buckets with missing fields',
+      }),
+    },
+    missingBucketLabel: {
+      types: ['string'],
+      default: i18n.translate('data.search.aggs.buckets.terms.missingBucketLabel', {
+        defaultMessage: 'Missing',
+        description: `Default label used in charts when documents are missing a field.
+      Visible when you create a chart with a terms aggregation and enable "Show missing values"`,
+      }),
+      help: i18n.translate('data.search.aggs.buckets.terms.missingBucketLabel.help', {
+        defaultMessage: 'Default label used in charts when documents are missing a field.',
+      }),
+    },
+    otherBucket: {
+      types: ['boolean'],
+      default: false,
+      help: i18n.translate('data.search.aggs.buckets.terms.otherBucket.help', {
+        defaultMessage: 'When set to true, groups together any buckets beyond the allowed size',
+      }),
+    },
+    otherBucketLabel: {
+      types: ['string'],
+      default: i18n.translate('data.search.aggs.buckets.terms.otherBucketLabel', {
+        defaultMessage: 'Other',
+      }),
+      help: i18n.translate('data.search.aggs.buckets.terms.otherBucketLabel.help', {
+        defaultMessage: 'Default label used in charts for documents in the Other bucket',
+      }),
+    },
+    exclude: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.buckets.terms.exclude.help', {
+        defaultMessage: 'Specific bucket values to exclude from results',
+      }),
+    },
+    include: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.buckets.terms.include.help', {
+        defaultMessage: 'Specific bucket values to include in results',
+      }),
+    },
+    json: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.buckets.terms.json.help', {
+        defaultMessage: 'Advanced json to include when the agg is sent to Elasticsearch',
+      }),
+    },
```

Comment:
I forgot that all agg types can have the optional advanced json. I didn't see a way around making this a string, unless we had a function run in a subexpression that literally just did `JSON.parse()`, e.g.

```
// json returns parsed `{ foo: true }`, or throws
aggTerms field="bytes" json={json "{ \"foo\": true }"}
```

Reply:
i think this is fine, the argument is named json, so its fine to pass json in i think.